src/dbnode/client/config.go

```diff
@@ -392,9 +393,11 @@ func (c Configuration) NewAdminClient(
 		encodingOpts = encoding.NewOptions()
 	}
 
-	v = v.SetReaderIteratorAllocate(func(r io.Reader, _ namespace.SchemaDescr) encoding.ReaderIterator {
-		intOptimized := m3tsz.DefaultIntOptimizationEnabled
-		return m3tsz.NewReaderIterator(r, intOptimized, encodingOpts)
+	v = v.SetReaderIteratorAllocate(func(r io.Reader, descr namespace.SchemaDescr) encoding.ReaderIterator {
+		if c.Proto != nil && c.Proto.Enabled {
```

Comment:
Hm instead of this if/else here, can you just entirely override with:
```go
v = v.SetReaderIteratorAllocate(func(r io.Reader, descr namespace.SchemaDescr) encoding.ReaderIterator {
  return proto.NewIterator(r, descr, encodingOpts)
})
```

From within the block below?
```go
if c.Proto != nil && c.Proto.Enabled {
  v = v.SetXyz(...)
 // ..
}
```

Reply:
Sure.